packages/sdk/src/oracle/client.ts

```diff
@@ -0,0 +1,91 @@
+import Store, { Emit } from "./store";
+import type { state } from "./types";
+import { Inputs } from "./types/state";
+import { Signer } from "./types/ethers";
+
+export class Client {
+  public readonly update: Update;
+  constructor(public readonly store: Store) {
+    this.update = new Update(store);
+  }
+  setUser(address: string, chainId: number, signer: Signer): void {
+    this.store.write((write) => write.user().set({ address, chainId, signer }));
+  }
+  setActiveRequest(params: Inputs["request"]): void {
+    const { requester, identifier, timestamp, ancillaryData, chainId } = params;
+    this.store.write((write) => write.inputs().request(requester, identifier, timestamp, ancillaryData, chainId));
+  }
+}
+
+export class Update {
```

Comment:
update functions are attached to the client, this allows fetching from chain and updating store